dashboard/src/api/Tree.tsx

```diff
@@ -9,9 +11,31 @@ const fetchTreeCheckoutData = async (origin: string): Promise<Tree[]> => {
   return res.data;
 };
 
-export const useTreeTable = (origin: string): UseQueryResult<Tree[]> => {
+export const useTreeTable = ({
+  enabled,
+}: {
+  enabled: boolean;
+}): UseQueryResult<Tree[]> => {
+  const { origin } = useSearch({ from: '/tree' });
```

Comment:
IMO it would be better to receive the origin as an argument rather than to obtain it itself, that way its more clear what values is being used in the query

Reply:
this query is specific to for tree, there are plenty of places to see what is being passed to the query including the network tab

Reply:
also, the hook is already an abstraction over the fetch request